spec/controllers/form_drafts_controller_spec.rb

```diff
@@ -232,6 +232,53 @@
     end
   end
 
+  describe 'POST #create' do
+    let :submit do
+      post :create, form_draft: { name: 'test' }, commit: @commit
+    end
+    context 'not staff' do
+      before :each do
+        when_current_user_is :not_staff
+      end
+      it 'does not allow access' do
+        expect_any_instance_of(FormDraft).not_to receive :create
+        submit
+        expect(response).to have_http_status :unauthorized
+      end
+    end
+    context 'staff' do
+      before :each do
+        when_current_user_is :staff
+      end
+      context 'commit is Save changes and continue editing' do
+        before :each do
+          @commit = 'Save changes and continue editing'
+        end
+        it 'creates a form with the changes given' do
+          expect { submit }.to change { Form.count }.by 1
+          expect(assigns[:draft].form.name).to eql 'test'
```

Comment:
I prefer `fetch` here, so that the exception is a little more helpful if this instance variable isn't assigned. With square brackets you get 'undefined method `form` for `nil:NilClass`, but with `fetch` you get 'Key not found: `draft`', which I think is more communicative.